### PR TITLE
Balances the masks of transmogriphication

### PR DESCRIPTION
**Pull request by MadmanMartian**
Masks take 5 seconds to put on.

Can no longer become boss mobs through the amorphous mask.

:cl:
* tweak: The magic of the masks of morphing falters, they are no longer as effective as they once were.

### CODE REVIEW

**Comment by Fruktlimpa** [comment]
Upset Xenoarch exploiters in 3..2..1

**Comment by Shadowmech88** [comment]
>Masks take 5 seconds to put on.

If this is because of people dying as their mob, then immediately taking the mask off and putting it back on again to re-fill its health, that could be fixed by just re-opening #14599.

**Comment by MadmanMartian** [comment]
It was more of a pre-warning that the nerd is about to become mecha hitler in a crowded bar, or gives people a chance to bully the nerd when he's refreshing his mask mob

**Comment by Shadowmech88** [comment]
What about tying the time necessary to the health of the mob in the mask somehow? 5 seconds seems like a long time to wait every time you want to turn into a corgi or something similarly weak.

File: code/modules/clothing/masks/transmog.dm [hunk 1]
```diff
@@ -124,8 +130,7 @@
 /obj/item/clothing/mask/morphing/amorphous/New()
 	..()
 	color = rgb(rand(0,255),rand(0,255),rand(0,255))
-	//Remove cockatrices because they're somewhat OP when player controlled
-	target_type = pick(existing_typesof(/mob/living/simple_animal) - existing_typesof(/mob/living/simple_animal/hostile/humanoid) - typesof(/mob/living/simple_animal/hostile/retaliate/cockatrice) - typesof(/mob/living/simple_animal/hostile/giant_spider/hunter/dead) - typesof(/mob/living/simple_animal/hostile/asteroid/hivelordbrood))
+	target_type = pick(existing_typesof(/mob/living/simple_animal) - existing_typesof(/mob/living/simple_animal/hostile/humanoid) - typesof(/mob/living/simple_animal/hostile/retaliate/cockatrice) - typesof(/mob/living/simple_animal/hostile/giant_spider/hunter/dead) - typesof(/mob/living/simple_animal/hostile/asteroid/hivelordbrood) - boss_mobs)
```
Comment (jknpj) [inline code/modules/clothing/masks/transmog.dm]: remove the manual cockatrice list and you might as well use the blacklisted mob list

**Review by jknpj — DISMISSED** [review]

**Comment by Blithering** [comment]
Maybe have dying when transformed drain the mask of power, giving it a cooldown before it can be put back on?

**Comment by PublicEnemyMaiPai** [comment]
What about removing the mask giving the user a seizure?

**Comment by Shadowmech88** [comment]
@Blithering 
Once #18206 is merged, dying while transformed will kill you for real.
Unless you think it should not do that, but in that case you should say so there.

File: code/modules/clothing/masks/transmog.dm [hunk 2]
```diff
@@ -26,6 +26,12 @@
 	if(cursed)
 		name = "cursed [name]"
 
+/obj/item/clothing/mask/morphing/mob_can_equip(mob/M, slot, disable_warning = 0, automatic = 0)
+	M.visible_message("<span class = 'warning'>\The [M] attempts to put on \the [src].</span>")
+	if(do_after(M, src, 50))
```
Comment (ShiftyRail) [inline code/modules/clothing/masks/transmog.dm]: 5 SECONDS reads better than 50 (deciseconds) - at least in my opinion.

**Comment by Blithering** [comment]
@Shadowmech88  Seems fair, so long as the mobs that tend to immediately die have been weeded out of the lists

**Comment by iFlashYou** [comment]
More science nerfs...

File: code/modules/clothing/masks/transmog.dm [hunk 3]
```diff
@@ -124,8 +130,7 @@
 /obj/item/clothing/mask/morphing/amorphous/New()
 	..()
 	color = rgb(rand(0,255),rand(0,255),rand(0,255))
-	//Remove cockatrices because they're somewhat OP when player controlled
-	target_type = pick(existing_typesof(/mob/living/simple_animal) - existing_typesof(/mob/living/simple_animal/hostile/humanoid) - typesof(/mob/living/simple_animal/hostile/retaliate/cockatrice) - typesof(/mob/living/simple_animal/hostile/giant_spider/hunter/dead) - typesof(/mob/living/simple_animal/hostile/asteroid/hivelordbrood))
+	target_type = pick(existing_typesof(/mob/living/simple_animal) - existing_typesof(/mob/living/simple_animal/hostile/humanoid) - typesof(/mob/living/simple_animal/hostile/giant_spider/hunter/dead) - typesof(/mob/living/simple_animal/hostile/asteroid/hivelordbrood) - boss_mobs - blacklisted_mobs)
```
Comment (jknpj) [inline code/modules/clothing/masks/transmog.dm]: ackshually i forgot blacklisted_mobs is a simple parent list, you'll need to use existing_typesof() on it

**Review by jknpj — CHANGES_REQUESTED** [review]

**Comment by DamianX** [comment]
when

**Comment by Pathid** [comment]
Five seconds do_after for putting it on is pretty anti-fun. Why not have it on the revert instead and add a cooldown to that so they're stuck as the animal for a while? Since #18206 got merged there's risk/reward there.

**Comment by DamianX** [comment]
5 seconds seems harsh but I don't like them doing it instantly either. Let's see how it goes, might tone it down at a later time.

**Pull request merged**